plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -623,4 +623,14 @@ private static void validateOrcBloomFilterColumns(ConnectorTableMetadata tableMe
             throw new TrinoException(INVALID_TABLE_PROPERTY, format("Orc bloom filter columns %s not present in schema", Sets.difference(ImmutableSet.copyOf(orcBloomFilterColumns), allColumns)));
         }
     }
+
+    public static String removeRedundantSlash(String location)
```

Comment:
"Remove redundant slash" sounds a bit  like "remove trailing slash" -- some operation that you do when manipulating paths.
Let's have the name express the fact this is a workaround for bad writers, not something you'd normally have to do.

`fixBrokenMetadataLocation`